docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -1,420 +1,125 @@
+    #[constructor]
+    fn constructor(
+        self: @ContractState,
+        initial_supply: u256,
+        recipient: ContractAddress
+    ) {
+        let name = 'MyToken';
+        let symbol = 'MTK';
 
-The following contract presets are ready to deploy and can be used as-is for quick prototyping and testing.
-Each preset mints an initial supply which is especially necessary for presets that do not expose a `mint` method.
-
-=== ERC20 (basic)
-
-The https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/token/erc20/presets/ERC20.cairo[`ERC20`] preset offers a quick and easy setup for deploying a basic ERC20 token.
-
-=== ERC20Mintable
-
-The https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/token/erc20/presets/ERC20Mintable.cairo[`ERC20Mintable`] preset allows the contract owner to mint new tokens.
-
-=== ERC20Pausable
-
-The https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/token/erc20/presets/ERC20Pausable.cairo[`ERC20Pausable`] preset allows the contract owner to pause/unpause all state-modifying methods i.e.
-`transfer`, `approve`, etc.
-This preset proves useful for scenarios such as preventing trades until the end of an evaluation period and having an emergency switch for freezing all token transfers in the event of a large bug.
-
-=== ERC20Upgradeable
-
-The https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/token/erc20/presets/ERC20Upgradeable.cairo[`ERC20Upgradeable`] preset allows the contract owner to upgrade a contract by deploying a new ERC20 implementation contract while also maintaining the contract's state.
-This preset proves useful for scenarios such as eliminating bugs and adding new features.
-For more on upgradeability, see xref:proxies.adoc#contract_upgrades[Contract upgrades].
-
-== API Specification
-
-=== Methods
-
-[,cairo]
-----
-func name() -> (name: felt) {
-}
-
-func symbol() -> (symbol: felt) {
-}
-
-func decimals() -> (decimals: felt) {
-}
-
-func totalSupply() -> (totalSupply: Uint256) {
-}
-
-func balanceOf(account: felt) -> (balance: Uint256) {
-}
-
-func allowance(owner: felt, spender: felt) -> (remaining: Uint256) {
-}
-
-func transfer(recipient: felt, amount: Uint256) -> (success: felt) {
+        let mut unsafe_state = ERC20::unsafe_new_contract_state();
+        ERC20::InternalImpl::initializer(ref unsafe_state, name, symbol);
+        ERC20::InternalImpl::_mint(ref unsafe_state, recipient, initial_supply);
+    }
 }
 
-func transferFrom(sender: felt, recipient: felt, amount: Uint256) -> (success: felt) {
-}
+    #[external(v0)]
+    impl MyTokenImpl of IERC20<ContractState> {
+        fn name(self: @ContractState) -> felt252 {
+            let mut unsafe_state = ERC20::unsafe_new_contract_state();
+            ERC20::ERC20Impl::name(ref unsafe_state, name, symbol); 
+        }
 
-func approve(spender: felt, amount: Uint256) -> (success: felt) {
+        ...
```

Comment:
```suggestion
        (...)
```

I think this is easier to read, and I've been using this in the code examples. If you think is better without parenthesis, let me know to update in my side.

Reply:
cc @martriay

Reply:
I lean toward no parenthesis but no strong opinions if we want to include them

Reply:
+1 to no parenthesis, +1 no strong opinion

Reply:
I ended up adding in the parenthesis because they do look a little naked with all empty space

Reply:
Just to confirm, should we use them then, right?

Reply:
Yes, sir. At least for now